web/src/components/graph/graphhoverline.jsx

```diff
@@ -15,8 +15,11 @@ const GraphHoverLine = React.memo(
     const layer = layers && layers[selectedLayerIndex];
     const fill = layer && layer.markerFill;
     const datapoint = layer && layer.datapoints && layer.datapoints[selectedTimeIndex];
-
-    const x = datetimes && datetimes[selectedTimeIndex] && timeScale(datetimes[selectedTimeIndex]);
+    const interval = datetimes && datetimes.length > 1 ? timeScale(datetimes[1]) - timeScale(datetimes[0]) : undefined;
+    let x = datetimes && datetimes[selectedTimeIndex] && timeScale(datetimes[selectedTimeIndex]);
+    if (interval) {
+      x += 0.5 * interval;
+    }
```

Comment:
The vertical line is just shifted by half an interval so it appears in the middle of it.